File: tests/human/x/zig/bool_chain.zig
const std = @import("std");

fn boom() bool {
    std.debug.print("boom\n", .{});
    return true;
}

pub fn main() void {
    std.debug.print("{}\n", .{(1 < 2) && (2 < 3) && (3 < 4)});
    std.debug.print("{}\n", .{(1 < 2) && (2 > 3) && boom()});
    std.debug.print("{}\n", .{(1 < 2) && (2 < 3) && (3 > 4) && boom()});
}

File: tests/human/x/go/bool_chain.go
//go:build ignore

package main

import "fmt"

func boom() bool {
	fmt.Println("boom")
	return true
}

func main() {
	fmt.Println((1 < 2) && (2 < 3) && (3 < 4))
	fmt.Println((1 < 2) && (2 > 3) && boom())
	fmt.Println((1 < 2) && (2 < 3) && (3 > 4) && boom())
}
